docs/proposals/version_documentation.md

```diff
@@ -0,0 +1,91 @@
+2. A solid and robust Documentation structure.
+3. Design a workflow for managing docs that integrates with Thanos Git workflow ie updating corresponding docs on pull requests, cherry-picks, etc.
+
+## Proposed Solution
+
+### Version Picker
+
+Currently, the documentation is residing under the docs/ folder of Thanos repository. It is built by Hugo. It will have a proper drop-down menu just like prometheus [`drop-down menu`](https://prometheus.io/docs/introduction/overview/) which will enable proper versioning. This user facing tool will be built using HTML and CSS.
+
+### Documentation Structure
+
+During the course of my research, I was able to deduce three different approaches of which we can solve this documentation structure issue.
+
+1. Sub branching: Creating different sub branches in the content folder of the Thanos repository. Since the current architecture of Hugo is this:
+
+|- archetypes
+|- config.toml
+|- content
+|- data
+|- layouts
+|- static
+|- themes
+
+You can add aditional versioned content in different folders within the content directory. For example:
+
+content
+|- docs
+|- master
+|- docs-v2
+|- docs-v3
+|- other-folder-for-other-releases
+
+_Note: The current version lives in the `docs` folder_
+
+#### Example:
+
+For `master` version we would be able to use the above links via the following path
+
+```
+/docs/master/components/
+```
+
+And for a minor version like `1.x`:
+
+```
+/docs/docs-v2/components/
+```
+
+2. Version Branching: You can decide to make each individual version to be a separate branch of it's own in the master repository.
+
+3. We can as well use templates/themes that have these processes automated e.g docsy, docusaurus, etc
+
+### Workflow
+
+Choosing a workflow solely depends on the kind of documentation structure we implement in our repositoory. The Sub branching documentation structure, which was listed above, can utilize the central workflow of which everything is tied to the master repository. You can make changes by pulling from the master, edit, and push new changes back to master. We can as well decide to implement the Version Branching workflow. In this workflow, You make each individual version to be a separate branch of it's own in the master repository. So each time a particular version of docs is selected from the version picker tool, it loads that particular branch.
+
+We can as well implement the Automated Scripting workflow. This allows us to use an automated script to get each content from each individual branch, and merge it to the latest branch.
+
+### Building a versioning plugin/extension
+
+Creating a plugin that can automate these process would save us alot of development time and stress. This approach promises to be very useful when it comes to versioning different release in the Thanos website. This can be achieved by implementing any of the following possible approach:
```

Comment:
Quick question, where there open source alternatives (plugins) where considered? if yes could you please list them and say why they wouldn't work. Maybe we can improve some other plugin instead of making our own?

Reply:
@povilasv During the course of my research, I wasn't able to find any open source alternatives (plugins) that could get the job done.

Although I found out that there are a couple of Hugo themes (such as Docsy and docusaurus ) that has scripts which automate the docs versioning process. These aforementioned themes were specifically built to support docs versioning and are quite used by alot of open source organisations.

I'm still doing more research to find out if there's an open source alternative (plugin) out there. If there exist a similar plugin, then I strongly believe it's best we improve it instead of making our own.

Reply:
I see thank you! Maybe add this comment to your proposal? This will help people understand why are we doing this ;)

Reply:
Alright. Would definitely do that. Thanks alot @povilasv